File: coderbyte/a_easy/javascript/16_arith_geo/arith_geo.js
const arithGeo = arr => {
  const arith = [];
  const geo = [];

  for (i = 0; i < arr.length - 1; i++) {
    arith.push(arr[i + 1] - arr[i]);
    geo.push(arr[i + 1] / arr[i]);
  }

  const pickUniqueValues = (value, index, self) => {
    return self.indexOf(value) === index;
  };

  if (arith.filter(pickUniqueValues).length === 1) {
    return 'Arithmetic';
  } else if (geo.filter(pickUniqueValues).length === 1) {
    return 'Geometric';
  } else {
    return -1;
  }
};

File: coderbyte/a_easy/ruby/16_arith_geo/arith_geo.rb
def arith_geo(arr)
  arith = []
  geo = []

  [*(0..arr.length - 2)].each do |n|
    arith << arr[n + 1] - arr[n]
    geo << arr[n + 1] / arr[n]
  end

  if arith.uniq.length == 1
    'Arithmetic'
  elsif geo.uniq.length == 1
    'Geometric'
  else
    -1
  end
end
